safe_transaction_service/history/tasks.py

```diff
@@ -395,12 +397,19 @@ def process_decoded_internal_txs_for_safe_task(
                 index_service.reprocess_addresses([safe_address])
 
             # Use iterator for memory issues
-            internal_txs_decoded = InternalTxDecoded.objects.pending_for_safe(
+            internal_txs_decoded_queryset = InternalTxDecoded.objects.pending_for_safe(
                 safe_address
-            ).iterator()
-            number_processed = len(
-                tx_processor.process_decoded_transactions(internal_txs_decoded)
             )
+            number_processed = 0
+            internal_txs_decoded_paginator = Paginator(
```

Comment:
You can set a chunk size for the iterator. Wouldn't that be sufficient?

Reply:
No, as it will try to process all the elements together